OnlineShop/OnlineShop.DAL/Exceptions/EntityNotFoundException.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OnlineShop.DAL.Exceptions
+{
+    public class EntityNotFoundException : RepositoryException
```

Comment:
You need to remove exceptions from this layer, they should be on the business logic layer